HLTrigger/HLTfilters/plugins/L1TTkMuonFilter.cc

```diff
@@ -145,23 +147,33 @@ L1TTkMuonFilter::MuonQualityCut::MuonQualityCut(const edm::ParameterSet& iConfig
   }
 }
 
-bool L1TTkMuonFilter::MuonQualityCut::operator()(const l1t::TkMuon& muon) const {
-  const auto& qualities = allowedQualities_.find(muon.muonDetector());
-  if (qualities != allowedQualities_.end()) {
-    return std::binary_search(qualities->second.begin(), qualities->second.end(), muon.quality());
-  } else {
-    return true;  //if qualities for that detector is not specified, we return true
+bool L1TTkMuonFilter::MuonQualityCut::operator()(const l1t::TrackerMuon& muon) const {
+  bool passesQuality(false);
+
+  // If we didn't load any qualities from the config file, that means we don't care.
+  // So we set passesQuality to true.
+  if (allowedQualities_.empty()) {
+    passesQuality = true;
+    return passesQuality;
   }
+
+  // The muonDetector() method is not available for TrackerMuon (it was available in TkMuon),
+  // so for the time being we just take the union of all quality vectors for all muon detectors.
+  for (const auto& detQuality : allowedQualities_) {
+    passesQuality = std::binary_search(detQuality.second.begin(), detQuality.second.end(), muon.hwQual());
```

Comment:
Shouldn't this be `passesQuality |=` ? Otherwise, `passesQuality` gets set only based on the last entry of `allowedQualities_`? If so, you can also add `if(passesQuality) break;` inside the look.

Reply:
Actually, this is messy because in principle the end user does not need to do anything with the qualities anymore. But I don't want to change the interface of the module, so we have to do something. I like the idea of using `passesQuality |=`.